flood_adapt/object_model/hazard/hazard.py

```diff
@@ -128,78 +177,260 @@ def add_discharge(self):
     @staticmethod
     def get_event_object(event_path):  # TODO This could be used above as well?
         mode = Event.get_mode(event_path)
-        if mode == "single_scenario":
+        if mode == "single_event":
             # parse event config file to get event template
             template = Event.get_template(event_path)
             # use event template to get the associated event child class
             return EventFactory.get_event(template).load_file(event_path)
-        elif mode == "probabilistic_set":
-            return None  # TODO: add Ensemble.load()
+        elif mode == "risk":
+            return EventSet.load_file(event_path)
+
+    def preprocess_models(self):
+        # Preprocess all hazard model input
+        self.preprocess_sfincs()
+        # add other models here
+
+    def postprocess_models(self):
+        # Preprocess all hazard model input
+        self.postprocess_sfincs()
+        # add other models here
 
     def run_models(self):
```

Comment:
shouldn't there be a seperate run_sfincs method that is run here? The same way you do for preprocess for example